app.py

```diff
@@ -49,61 +52,92 @@ def login():
         return jsonify(user={'username': 'test'}, is_authenticated=True), 200
 
 
-@app.route('/api/chats', methods=['GET', 'POST'])
-def chats():
-    """
-    if method == GET
-    :return: All chats
+class Chats(Resource):
 
-    if method == POST
-    send user id and chat name
-    :return:
-    """
-    if request.method == 'GET':
+    def get(self):
         chats = [
             {"id": 1, "chat_name": "skiribops"},
             {"id": 2, "chat_name": "Subscribe to PewDiePie"},
             {"id": 3, "chat_name": "DB"}
         ]
-        return jsonify(results=chats), 200
-
-    elif request.method == 'POST':
-        chat_name = request.form['chat_name']
-        owner_id = request.form['uid']
-        return jsonify(chat={'id': 2, 'chat_name': chat_name}, msg='Success'), 201
-
-
-@app.route('/api/chats/<int:cid>', methods=['GET'])
-def chatByID(cid):
-    if request.method == 'GET':
-        return jsonify(chat={'id': cid, 'chat_name': 'skiribops'}), 200
-
-
-@app.route('/api/contacts/<int:uid>', methods=['GET', 'POST'])
-def contacts(uid):
-    """
-    if GET
-    :param uid: user id de quien quieres la lista
-    :return:
-
-    if POST
-    add a user to the contact del uid en el url
-    """
-    if request.method == 'GET':
-        return jsonify(contacts=[{'uid': 3}, {'uid': 4}]), 200
-    else:
-        uid_to_add = request.form['uid']
-        return jsonify(msg='added'), 201
-
-
-@app.route('/chats/<int:cid>/messages')
-def messages():
-    if request.method == 'GET':
-        return jsonify()
-    else:
-        # add message to chat
-        pass
-
+        return jsonify(chats=chats)
+
+    def post(self):
+        chat = {'id': 2, 'chat_name': 'Videout'}
+        return jsonify(chat=chat, msg='Success')
+
+
+class Chat(Resource):
+
+    def get(self, chat_id):
+        chat = {
+            "text": "This is a test message!",
```

Comment:
Deberia devoler los atributos ya definidos en el "ER"

Reply:
This should return chat messages for the specific chat. However I'm not sure of the attributes. We should add the E-R diagram to the project

Reply:
Made some changes to this method. Review it once more